app/jobs/organization_notifier_job.rb

```diff
@@ -12,9 +12,19 @@ class OrganizationNotifierJob < ActiveJob::Base
   def perform
     Organization.all.find_each do |org|
       posts = org.posts.active.of_active_members.from_last_week
+
       if posts.present?
-        OrganizationNotifier.recent_posts(posts).deliver_now
+        users_by_locale(org).each do |locale, users|
+          OrganizationNotifier.recent_posts(posts, locale, users).deliver_now
+        end
       end
     end
   end
+
+  private
+
+  def users_by_locale(organization)
+    organization.users.online_active.actives.
+      notifications.group_by { |u| u.locale || I18n.default_locale }
```

Comment:
We already have the locale column defaulting to `es` so we can ask PostgreSQL to do the grouping for us, right? I'm just a bit concerned about the little visibility we have over the data fetching we do from this background job. It's the kind of thing we might realize too late that it's killing the DB.

Reply:
ah yes! that was my initial idea too (avoid in-memory sorting), but I thought that column doesn't have a default... but yes, you're right, it has a default in the schema (maybe I have some broken records in local DB), so we can rely on that.

Reply:
@sauloperez I'm just trying to move the grouping thing to postgres, but if I'm not wrong we'll need anyway some Ruby and in-memory operations to mount a hash or some data-structure like that:

```ruby
{
  es: [array of users with ES],
  en: [array of users with EN],
  ca: [array of users with CA]
  ...
}
```

Another way to implemente this, would be 1 query for each available locale in the given organization.

Reply:
I tried to a quick look at this last week and didn't get anything. The problem is that we don't want only the locales but also the users within each one and that's nothing we can get with `GROUP BY`. The only way would be with PostgreSQL window functions but it feels too much to stuff into ActiveRecord for this simple use case. A query per locale seems enough now.

Reply:
Yep agree, I see "feasible" 2 options:
- group_by: 1 query, but more memory/cpu consuming due to Ruby in-memory operations [current solution]
- multiple queries: 1 query x locale, but no in-memory sorting

Reply:
@sauloperez approach (2) pushed in 8f62ec6